zcash_proofs/src/prover.rs

```diff
@@ -60,6 +60,26 @@ impl LocalTxProver {
         }
     }
 
+    /// Attempts to create a `LocalTxProver` using parameters from specified as byte arrays.
+    #[cfg(feature = "local-prover")]
+    #[cfg_attr(docsrs, doc(cfg(feature = "local-prover")))]
+    pub fn new_with_params(spend_params: &[u8], output_params: &[u8]) -> Self {
+        // Deserialize params
+        let spend_params = Parameters::<Bls12>::read(spend_params, false)
+            .expect("couldn't deserialize Sapling spend parameters file");
```

Comment:
Could you please return `Result` or `Option` rather than panicing? The caller can panic, but this should be a pure function.

Reply:
@adityapk00 if you're able to resolve this in the next day or two, we'll be able to get this PR into the 4.1.0 release.

Reply:
Thanks, @nuttycom - I refactored it a bit, so it uses the same `load_parameters` method, which also checks the hashes.